tests/shared.py

```diff
@@ -186,10 +186,14 @@ def run(self, *command):
 
 
 class GitRepo(Repo):
-    def __init__(self, content_dir):
+    def __init__(self, content_dir, **init_default_branch):
         super().__init__(content_dir)
 
         self.run('git', 'init')
+        # includes option to change the initial default branch
+        if 'default_branch' in init_default_branch:
+            self.run('git', 'symbolic-ref',
+            'HEAD', 'refs/heads/' + init_default_branch['default_branch'])a
```

Comment:
What's the `a` doing at the end of this line? I feel like that should be a syntax error? That might be your test failure. One way to check this is to introduce some HUGE syntax error and see if you get a different failure. If not, then your current failure is probably a syntax error :)

Reply:
Rookie mistake, sorry.
Actually I removed this code from there, and it is still failing at the same test, and it is not an error.
I will keep investigating.

Reply:
I just removed the function from main() in git_plugin.py and the failures is gone.
Is `def test_git_plugin_multiple_fetches(self)` doing something different that could interfere?
I still do not completely understand the testing functions.

Reply:
Somehow, running this function (git_default_branch) before the `plugin_sync` or `plugin_reup` functions is causing the test to fail. 
The only thing that touches another part is `clone_if_needed(),` so it made me think that running this `clone_if_needed() `before is causing something to collapse in the multiple fetches test. Any clues?

Reply:
Yay, with the removal of the function `clone_maybe_print()` the error in the test is gone.